lib/elixir/lib/file/stream.ex

```diff
@@ -115,8 +116,14 @@ defmodule File.Stream do
       {:error, __MODULE__}
     end
 
+    defp strip_bom(device) do
+      header = IO.binread(device, 4)
+      {:ok, _new_pos} = :file.position(device, File.bom_length(header))
+      device
+    end
+
     defp read_modes(modes) do
-      for mode <- modes, mode not in [:write, :append], do: mode
+      for mode <- modes, mode not in [:write, :append, :strip_bom], do: mode
```

Comment:
I would personally prefer `Enum.uniq(modes) -- [:write, :append, :strip_bom]`

A version with just `--` would be ideal, but there's a possibility of duplicate entries.